autopush/tests/test_db.py

```diff
@@ -11,40 +10,37 @@
 from nose.tools import eq_
 
 from autopush.db import (
-    create_router_table,
-    create_storage_table,
     get_router_table,
     get_storage_table,
     Storage,
     Router,
 )
+from autopush.settings import MetricSink
 
 
-class StorageTestCase(unittest.TestCase):
-    def setUp(self):
-        self.mock_dynamodb2 = mock_dynamodb2()
-        self.mock_dynamodb2.start()
+mock_db2 = mock_dynamodb2()
 
-    def tearDown(self):
-        self.mock_dynamodb2.stop()
 
-    def test_storage_table_created(self):
-        # Check that the table doesn't exist
-        db = DynamoDBConnection()
-        dblist = db.list_tables()["TableNames"]
-        assert "storage" not in dblist
+def setUp():
+    mock_db2.start()
 
-        # Create the storage table
-        create_storage_table()
 
-        dblist = db.list_tables()["TableNames"]
-        assert "storage" in dblist
+def tearDown():
+    mock_db2.stop()
+
+
+class StorageTestCase(unittest.TestCase):
+    def setUp(self):
+        table = get_storage_table()
+        self.real_table = table
+        self.real_connection = table.connection
 
-        get_storage_table()
+    def tearDown(self):
+        self.real_table.connection = self.real_connection
```

Comment:
So, only the connection needs to be restored from my experiments, because mock_dynamodb2 will otherwise retain the one I messed up. The table is created fresh each instantiation, but mock_dynamodb2 retains the connection and re-assigns it to new table objects.